parser/charset/encoding_ascii.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package charset
+
+import (
+	"strings"
+	go_unicode "unicode"
+
+	"golang.org/x/text/encoding"
+)
+
+// EncodingASCIIImpl is the instance of EncodingASCII.
+var EncodingASCIIImpl = &EncodingASCII{
+	EncodingBase{enc: encoding.Nop},
+}
+
+// EncodingASCII is the ASCII encoding.
+type EncodingASCII struct {
+	EncodingBase
+}
+
+// ToUpper implements Encoding interface.
+func (e *EncodingASCII) ToUpper(src string) string {
+	return strings.ToUpper(src)
+}
+
+// ToLower implements Encoding interface.
+func (e *EncodingASCII) ToLower(src string) string {
+	return strings.ToLower(src)
+}
+
+// Name implements Encoding interface.
+func (e *EncodingASCII) Name() string {
+	return CharsetASCII
+}
+
+// Peek implements Encoding interface.
+func (e *EncodingASCII) Peek(src []byte) []byte {
+	if len(src) == 0 {
+		return src
+	}
+	return src[:1]
+}
+
+func (e *EncodingASCII) Validate(dest, src []byte) (result []byte, nSrc int, ok bool) {
```

Comment:
A bit weird in func name `Validate`.
At first sight, I thought it was just for testing whether `[]byte` is legal or not. But it also provides transfor ability. Maybe split it to `validate` and `transform` is better?

Reply:
How about changing its name to `EncodeAndDecode`?

Reply:
`TryReplaceInvalidChars` is much easier to understand.

Reply:
Updated. I add the function `ReplaceIllegal` to the interface. Now `Validate` only tests whether `[]byte` is legal.